src/frontend/test_runner/tests/testdata/distribution_derive.yaml

```diff
@@ -1,63 +1,429 @@
+      StreamExchange { dist: HashShard(ak1.a._row_id, a.k1) }
+        StreamHashJoin { type: Inner, predicate: ak1.k1 = a.k1, output: [ak1.v, sum(count), ak1.a._row_id, a.k1] }
+          StreamExchange { dist: HashShard(ak1.k1) }
+            StreamTableScan { table: ak1, columns: [ak1.k1, ak1.v, ak1.a._row_id], pk: [ak1.a._row_id], distribution: UpstreamHashShard(ak1.k1) }
+          StreamProject { exprs: [sum(count), a.k1] }
+            StreamHashAgg { group_key: [a.k1], aggs: [sum(count), sum(count)] }
+              StreamExchange { dist: HashShard(a.k1) }
+                StreamHashAgg { group_key: [a.k1, Vnode(a._row_id)], aggs: [count, count] }
+                  StreamProject { exprs: [a.k1, a._row_id, Vnode(a._row_id)] }
+                    StreamTableScan { table: a, columns: [a.k1, a._row_id], pk: [a._row_id], distribution: UpstreamHashShard(a._row_id) }
+- id: aggk1_join_Ak1_onk1
+  before:
+  - create_tables
+  sql: |
+    with b as (
+      select
+        count(*) as num, k1
+      from A
+      group by k1
+    )
+    select A.v, B.num as Bv from B join Ak1 as A using(k1)
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+      BatchHashJoin { type: Inner, predicate: a.k1 = ak1.k1, output: [ak1.v, count] }
+        BatchProject { exprs: [count, a.k1] }
+          BatchHashAgg { group_key: [a.k1], aggs: [count] }
+            BatchExchange { order: [], dist: HashShard(a.k1) }
+              BatchScan { table: a, columns: [a.k1], distribution: SomeShard }
+        BatchExchange { order: [], dist: HashShard(ak1.k1) }
+          BatchScan { table: ak1, columns: [ak1.k1, ak1.v], distribution: UpstreamHashShard(ak1.k1) }
+  stream_plan: |
+    StreamMaterialize { columns: [v, bv, a.k1(hidden), ak1.a._row_id(hidden)], pk_columns: [a.k1, ak1.a._row_id] }
+      StreamHashJoin { type: Inner, predicate: a.k1 = ak1.k1, output: [ak1.v, sum(count), a.k1, ak1.a._row_id] }
+        StreamProject { exprs: [sum(count), a.k1] }
+          StreamHashAgg { group_key: [a.k1], aggs: [sum(count), sum(count)] }
+            StreamExchange { dist: HashShard(a.k1) }
+              StreamHashAgg { group_key: [a.k1, Vnode(a._row_id)], aggs: [count, count] }
+                StreamProject { exprs: [a.k1, a._row_id, Vnode(a._row_id)] }
+                  StreamTableScan { table: a, columns: [a.k1, a._row_id], pk: [a._row_id], distribution: UpstreamHashShard(a._row_id) }
+        StreamExchange { dist: HashShard(ak1.k1) }
+          StreamTableScan { table: ak1, columns: [ak1.k1, ak1.v, ak1.a._row_id], pk: [ak1.a._row_id], distribution: UpstreamHashShard(ak1.k1) }
+- id: aggk1_join_aggk1_onk1
+  before:
+  - create_tables
+  sql: |
+    with a as (
+      select
+        count(*) as num, k1
+      from A
+      group by k1
+    ), b as (
+      select
+        count(*) as num, k1
+      from B
+      group by k1
+    )
+    select A.num, B.num as Bv from a join b using(k1)
```

Comment:
```suggestion
    select a.num, b.num as Bv from a join b using(k1)
```

Reply:
BTW, can `aggk1 as A join aggk1 as B`

Reply:
there is some bugs, I will fix it in another PR

Reply:
Why not fix them here 😢